docs/plugins/gatsby-remark-relative-linker/functions.js

```diff
@@ -0,0 +1,53 @@
+const SIBLING = "./";
+const PARENT = "../";
+const UNPREFIXED = -1;
+
+function* depthCheck(n) {
+  for (let i = n; i > -1; i--) yield i;
+}
+
+function escapeRegExp(string) {
+  //https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
+}
+
+function buildDepthCheckRegex(depth) {
+  if (depth === 0) return new RegExp("^" + escapeRegExp(SIBLING));
+  return new RegExp("^" + escapeRegExp(PARENT.repeat(depth)));
+}
+
+function doUpdateDepth(linkText, forDepth, isIndex = false) {
+  let noPrefixRegex = /^[\w-_]+/;
+
+  if (linkText.match(noPrefixRegex)) {
+    return updateLinkText(linkText, UNPREFIXED, isIndex);
+  }
+
+  let depthCheckRegex = buildDepthCheckRegex(forDepth);
+  let result = linkText.match(depthCheckRegex);
+
+  if (result) {
+    return updateLinkText(linkText, forDepth, isIndex);
+  }
+}
+
+function updateLinkText(linkText, depth, index) {
+  switch (index) {
+    case true:
+      return depth === UNPREFIXED ? SIBLING + linkText : linkText;
+    case false:
+      return PARENT + linkText;
+  }
+}
+
+function updateRelativeDepth(linkText, isIndex, MAX_DEPTH_CHECK = 6) {
+  for (const depth of depthCheck(MAX_DEPTH_CHECK)) {
+    const updatedLinkText = doUpdateDepth(linkText, depth, isIndex);
+    if (updatedLinkText) {
+      return updatedLinkText;
+    }
+  }
+  return linkText;
+}
+
+module.exports = updateRelativeDepth;
```

Comment:
I think this PR will work, but curious could this version also work? Basically I'm trying to understand the necessity of having the for-loop in line 44: is it true that regardless of the depth we want to check, as long as the link is inside a page that is NOT index.md or README.md, prefixing ../ would be all we need? Let me know if I missed anything 

```
const PARENT = "../";

function updateRelativeDepth(linkText, index) {
  switch (index) {
    case true:
      return linkText;
    case false:
      return PARENT + linkText;
  }
}

module.exports = updateRelativeDepth;
```

Reply:
lol yup, it looks like that will work. Removing all the code now.